evap/staff/views.py

```diff
@@ -462,6 +456,10 @@ def helper_course_edit(request, semester, course):
     if form.is_valid() and formset.is_valid():
         if operation not in ('save', 'approve'):
             raise SuspiciousOperation("Invalid POST operation")
+
+        if (not course.can_staff_edit or course.is_archived) and operation == 'save':
+            raise SuspiciousOperation("Modifying this course is not allowed.")
```

Comment:
i think this check should also apply for the `approve` operation, right? we don't want people to approve courses that they cannot edit.